### ISSUE TITLE
Crypting EC pkey with a user-supplied passphrase fails on 3.0.0-alpha16-dev

### ISSUE THREAD

**Issue by gotthardp** [openssl/openssl#15114]
On OpenSSL 1.1.1d this works
```
$ openssl ecparam -name secp384r1 -genkey | openssl ec -aes-256-cbc -out key.pem
read EC key
writing EC key
Enter PEM pass phrase:
Verifying - Enter PEM pass phrase:
```
However on OpenSSL 3.0.0-alpha16-dev I get an error
```
$ openssl ecparam -name secp384r1 -genkey | openssl ec -aes-256-cbc -out key.pem
read EC key
writing EC key
unable to write EC key
804780A5AA7F0000:error:078C0103:common libcrypto routines:ossl_pw_get_passphrase:internal error::0:
804780A5AA7F0000:error:0480006F:PEM routines:PEM_ASN1_write_bio:read key::0:
```

But when I supply the password from the command line, everything is fine even on the 3.0
```
$ openssl ecparam -name secp384r1 -genkey | openssl ec -aes-256-cbc -passout pass:secret -out key.pem
read EC key
writing EC key
```

**Comment by Jasonbuchanan1983** [comment]
https://slack.com/downloads

**Comment by levitte** [comment]
I just had a look at `openssl rsa` and `openssl dsa` while looking at this, and discovered that they completely ignore the key encryption settings completely.

Fix coming up!

**Comment by levitte** [comment]
Fix in #15119
